pkg/health/condition/check_ready.go

```diff
@@ -14,22 +14,26 @@
 
 package condition
 
-import druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+import (
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/utils"
+)
 
 type readyCheck struct{}
 
 func (r *readyCheck) Check(status druidv1alpha1.EtcdStatus) Result {
-	if len(status.Members) == 0 {
+	if status.ClusterSize == nil {
 		return &result{
 			conType: druidv1alpha1.ConditionTypeReady,
 			status:  druidv1alpha1.ConditionUnknown,
-			reason:  "NoMembersInStatus",
-			message: "Cannot determine readiness of cluster since status has no members",
+			reason:  "ClusterSizeUnknown",
+			message: "Cannot determine readiness of cluster since no cluster size has been calculated",
 		}
 	}
 
 	var (
-		quorum       = len(status.Members)/2 + 1
+		size         = utils.Max(int(*status.ClusterSize), len(status.Members))
```

Comment:
This could be now simply `int(*status.ClusterSize)`, right?

Reply:
Not if the cluster size changes at the same time. `status.ClusterSize` is only calculated in the status update [here](https://github.com/gardener/etcd-druid/pull/188/files#diff-a2ef4cdc737ef6465c303ae9670522738c19205fb68a412813727558230c43a3R141), after the conditions have been calculated.